project-koku/koku
# [COST-5170] - Auto large provider based on report count

## Jira Ticket

[COST-5170](https://issues.redhat.com/browse/COST-5170)

## Description

This change will add a small amount of logic to manifest processing to adjust the queue in which tasks land for the given provider based on whether its deemed large by report count.

## Testing

1. Checkout Branch
2. Restart Koku
3. Set `XL_REPORT_COUNT` to 0 (Or generate a large amount of report files with nise)
4. Load data
5. You should see the following log message during processing `marking provider large as report files exceed threshold.` also if you are quick or scale down your worker you can check to see that tasks are queued in the relevant XL queues.

Additionally ingest a provider with less reports than `XL_REPORT_COUNT` and see that we process like normal in the standard queue.

## Release Notes

```markdown
* [COST-5170](https://issues.redhat.com/browse/COST-5170) Add auto large provider logic.
```
